File: backend/app/schemas/market.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class MarketResponse(BaseModel):
    id: uuid.UUID
    condition_id: str
    question: str
    description: str | None
    category: str | None
    slug: str | None
    yes_price: Decimal | None
    no_price: Decimal | None
    volume: Decimal | None
    liquidity: Decimal | None
    end_date: datetime | None
    is_active: bool
    resolved: bool
    last_fetched_at: datetime

    model_config = {"from_attributes": True}


class EnhancedMarketResponse(MarketResponse):
    edge_potential: float | None = None
    liquidity_score: float | None = None
    composite_score: float | None = None
    research_status: str | None = None
    last_researched_at: datetime | None = None
    user_has_position: bool = False
    position_side: str | None = None
    spread: float | None = None
    hours_to_resolution: float | None = None

File: frontend/src/types/market.ts
export interface EnhancedMarket {
  id: string;
  condition_id: string;
  question: string;
  description: string | null;
  category: string | null;
  slug: string | null;
  yes_price: number | null;
  no_price: number | null;
  volume: number | null;
  liquidity: number | null;
  end_date: string | null;
  is_active: boolean;
  resolved: boolean;
  last_fetched_at: string;
  edge_potential: number | null;
  liquidity_score: number | null;
  composite_score: number | null;
  research_status: 'researched' | 'stale' | null;
  last_researched_at: string | null;
  user_has_position: boolean;
  position_side: string | null;
  spread: number | null;
  hours_to_resolution: number | null;
}
